src/core/cgroup.c

```diff
@@ -189,6 +190,268 @@ void cgroup_context_init(CGroupContext *c) {
+        dst->default_memory_low = src->default_memory_low;
+        dst->default_startup_memory_low = src->default_startup_memory_low;
+        dst->memory_min = src->memory_min;
+        dst->memory_low = src->memory_low;
+        dst->startup_memory_low = src->startup_memory_low;
+        dst->memory_high = src->memory_high;
+        dst->startup_memory_high = src->startup_memory_high;
+        dst->memory_max = src->memory_max;
+        dst->startup_memory_max = src->startup_memory_max;
+        dst->memory_swap_max = src->memory_swap_max;
+        dst->startup_memory_swap_max = src->startup_memory_swap_max;
+        dst->memory_zswap_max = src->memory_zswap_max;
+        dst->startup_memory_zswap_max = src->startup_memory_zswap_max;
+
+        dst->default_memory_min_set = src->default_memory_min_set;
+        dst->default_memory_low_set = src->default_memory_low_set;
+        dst->default_startup_memory_low_set = src->default_startup_memory_low_set;
+        dst->memory_min_set = src->memory_min_set;
+        dst->memory_low_set = src->memory_low_set;
+        dst->startup_memory_low_set = src->startup_memory_low_set;
+        dst->startup_memory_high_set = src->startup_memory_high_set;
+        dst->startup_memory_max_set = src->startup_memory_max_set;
+        dst->startup_memory_swap_max_set = src->startup_memory_swap_max_set;
+        dst->startup_memory_zswap_max_set = src->startup_memory_zswap_max_set;
+
+        SET_FOREACH(i, src->ip_address_allow) {
+                r = in_addr_prefix_add(&dst->ip_address_allow, i);
+                if (r < 0)
+                        return -ENOMEM;
+        }
+
+        SET_FOREACH(i, src->ip_address_deny) {
+                r = in_addr_prefix_add(&dst->ip_address_deny, i);
+                if (r < 0)
+                        return -ENOMEM;
+        }
+
+        dst->ip_address_allow_reduced = src->ip_address_allow_reduced;
+        dst->ip_address_deny_reduced = src->ip_address_deny_reduced;
+
+        if (!strv_isempty(src->ip_filters_ingress)) {
+                dst->ip_filters_ingress = strv_copy(src->ip_filters_ingress);
+                if (!dst->ip_filters_ingress)
+                        return -ENOMEM;
+        }
+
+        if (!strv_isempty(src->ip_filters_egress)) {
+                dst->ip_filters_egress = strv_copy(src->ip_filters_egress);
+                if (!dst->ip_filters_egress)
+                        return -ENOMEM;
+        }
+
+        LIST_FOREACH(programs, l, src->bpf_foreign_programs) {
+                _cleanup_free_ CGroupBPFForeignProgram *n = NULL;
+
+                n = new0(CGroupBPFForeignProgram, 1);
+                if (!n)
+                        return -ENOMEM;
+
+                n->bpffs_path = strdup(l->bpffs_path);
+                if (!n->bpffs_path)
+                        return -ENOMEM;
+
+                n->attach_type = l->attach_type;
+
+                LIST_PREPEND(programs, dst->bpf_foreign_programs, TAKE_PTR(n));
```

Comment:
you are basically duplicating cgroup_context_add_bpf_foreign_program() here.

please add a helper cgroup_context_add_bpf_foreign_program_dup() or so which is a wrapper around cgroup_context_add_bpf_foreign_program() and duplicates an existing CGroupBBPFForeignProgram

Reply:
Hmm, `cgroup_context_add_bpf_foreign_program()` already duplicated its arguments so why not just call it with `l->bpffs_path` and `l->attach_type`?